pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -557,6 +596,94 @@ func (ku *Upgrader) generateUpgradeTemplate(upgradeContainerService *api.Contain
 	return templateMap, parametersMap, nil
 }
 
+func (ku *Upgrader) getLastVMNameInVMSS(ctx context.Context, resourceGroup string, vmScaleSetName string) (string, error) {
```

Comment:
Can we unit test this?

Reply:
very like to add an UT to cover this. But ListVirtualMachineScaleSetVMs() in MockAKSEngineClient always return empty list. Not too helpful to UT this function. But I manually tested that this  function returns the last node name in an VMSS agent pool.

Reply:
@serbrech do you have any guidance for Charlie here for following what you did to test your own upgrade changes? I would really like us to avoid adding new untested code

Reply:
Yes, you can see how it is setup here : 
you need to setup the ScaleSetResult first : https://github.com/Azure/aks-engine/blob/a5a95741849510f25f623fc2c792859e38f87199/pkg/operations/kubernetesupgrade/upgradecluster_test.go#L255

and then setup the list of VMs that you want to be returned : 
https://github.com/Azure/aks-engine/blob/a5a95741849510f25f623fc2c792859e38f87199/pkg/operations/kubernetesupgrade/upgradecluster_test.go#L276

Reply:
you can do the same with VMAS, but you just need to setup the list of VMs cause the api is simpler : 
https://github.com/Azure/aks-engine/blob/a5a95741849510f25f623fc2c792859e38f87199/pkg/operations/kubernetesupgrade/upgradecluster_test.go#L354

Reply:
After Stephane's commit today, it's easy to test getLastVMNameInVMSS. Thanks